test/bats/commands.bats

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env bats
+
+# Tests do not check output yet
+
+setup() {
+  if [[ -d test ]] ; then cd test ; fi
+  if [[ "$BATS_TEST_NUMBER" != "1" ]] && [[ -d myapp ]] ; then cd myapp ; fi
+}
+
+@test "Create app" {
+  [[ -d myapp ]] && rm -rf myapp
+  npx qx create myapp -I --type server -v
+  cd myapp
+  npx qx compile -v --clean
```

Comment:
Wouldn't `npx` look first in the *current* project first? This is why @oetiker recommends to always use `npx qx` - it wouldn't use a global qx, but instead the executable that is installed in `node_modules`, or - in our case - the one that in the current project itself. A quick test showed that if I do `npx qx --version` in the `qooxdoo-compile` folder, it will show the version `v1.0.0-beta` (which is the one of the project) and not `v1.0.0.beta.XXXXXXX`. which is the globally installed qx. But please correct me if I am wrong.

Reply:
https://www.npmjs.com/package/npx is ambiguous: 
> Executes <command> either from a local node_modules/.bin, or from a central cache, installing any packages needed in order for <command> to run.
> By default, npx will check whether <command> exists in $PATH, or in the local project binaries, and execute that. If <command> is not found, it will be installed prior to execution.
But from what I've read, it seems clear that npx has been written to priorities the *local* binary over the global. @oetiker, do you know more?

Reply:
it looks in the node_modules/.bin folder too. So if there exists a qx module - what's result's from a false Installation - this is eventually used. Therefore i prefer clear Tasks for tests

Reply:
It's impossible to install a module inside itself: 
```
qooxdoo-compiler $ npm install @qooxdoo/compiler
npm ERR! code ENOSELF
npm ERR! Refusing to install package with name "@qooxdoo/compiler" under a package
npm ERR! also called "@qooxdoo/compiler". Did you name your project the same
npm ERR! as the dependency you're installing?
```
Since the tests run only in the `@qooxdoo/compiler` package, and are mainly about travis, why should we worry about broken installs?

Reply:
ok. I had this problem during switch from qxcompiler. Somewhat academic.